.github/scripts/upload_test_stats_datadog.py

```diff
@@ -77,6 +79,11 @@ def get_workflow_runs_from_github() -> List[Dict]:
     while True:
         print("Fetching page", page, "...")
         response = requests.get(url + f"&page={page}", headers=HEADERS, timeout=30)
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as err:
+            logger.exception("An error occurred while processing the response: %s", err)
+            raise RuntimeError(f"An error occurred while processing the response: {err}") from err
```

Comment:
```suggestion
            raise
```

you don't have to raise a custom error, logging + re-raising the requests error can be enough

but I'm fine with the runtimeerror too